lib/attr_encrypted/adapters/active_record.rb

```diff
@@ -88,8 +99,9 @@ def method_missing_with_attr_encrypted(method, *args, &block)
             if match = /^(find|scoped)_(all_by|by)_([_a-zA-Z]\w*)$/.match(method.to_s)
               attribute_names = match.captures.last.split('_and_')
               attribute_names.each_with_index do |attribute, index|
-                if attr_encrypted?(attribute)
+                if attr_encrypted?(attribute) && encrypted_attributes[attribute.to_sym][:mode] == :single_iv_and_salt
                   args[index] = send("encrypt_#{attribute}", args[index])
+                  puts "DEPRECATION WARNING: This feature will be removed in the next major release."
```

Comment:
Would [`warn`](http://ruby-doc.org/core-2.3.0/Kernel.html#method-i-warn) be more appropriate here?